src/components/proposalBooking/CreateUpdateVisit.tsx

```diff
@@ -0,0 +1,129 @@
+import { Button, Typography } from '@material-ui/core';
+import { Form, Formik } from 'formik';
+import React from 'react';
+import * as Yup from 'yup';
+
+import { ActionButtonContainer } from 'components/common/ActionButtonContainer';
+import ErrorMessage from 'components/common/ErrorMessage';
+import FormikDropdown from 'components/common/FormikDropdown';
+import Participants from 'components/proposal/ProposalParticipants';
+import { BasicUserDetails } from 'generated/sdk';
+import useDataApiWithFeedback from 'utils/useDataApiWithFeedback';
+import { getFullUserName } from 'utils/user';
+
+import { ProposalScheduledEvent } from '../../hooks/proposalBooking/useProposalBookingsScheduledEvents';
+
+interface CreateUpdateVisitProps {
+  event: ProposalScheduledEvent;
+  close: (updatedEvent: ProposalScheduledEvent) => void;
+}
+function CreateUpdateVisit({ event, close }: CreateUpdateVisitProps) {
+  const { api } = useDataApiWithFeedback();
+
+  const visit = event.visit;
+
+  const initialValues = {
+    team: visit?.registrations.map((registration) => registration.user) || [],
+    teamLeadUserId: visit?.teamLead.id,
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={Yup.object().shape({
```

Comment:
I see you keep the validation schema here directly in the form. Not sure maybe it is better to move to the validation package because I see many tasks that say we should try to move all validations left in the frontend and backend in the validation package.

Reply:
Good idea, I'll make another PR for this change